### ISSUE TITLE
[Bug]: E2E Treats All Emitted Events as Anonymous (Ignores Event Names)

### ISSUE THREAD

**Issue by 0xNeshi** [OpenZeppelin/rust-contracts-stylus#408]
### What happened?

The problem was first noticed here:
https://github.com/OpenZeppelin/rust-contracts-stylus/pull/352#discussion_r1834604590

Explanation: we are trying to assert that the correct event was emitted when `Ownable2Step::transfer_ownership`  was called, which _should_ be `OwnershipTransferStarted`, but the wrong event was asserted to had been emitted (`OwnerhipTransferred`). **This assertion should have failed the test run, but it somehow passed!**
This is when we realized there might be a problem with our e2e tool.

After further investigation, the reason for this became evident - [currently used version of `alloy-sol-types` (0.7.6)](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/0815bdd5c9af8d1cd534e3719f9ccc2e73d92fd4/Cargo.toml#L85) does not check event signature when decoding logs, in effect treating all events as anonymous.

Pasting below the main part of the [investigation report](https://openzeppelin.slack.com/archives/C06R5AGFE9G/p1731533202511839):
> I think I have found the reason for `Ownable2Step` e2e bug.
> 
> Long story short, I managed to confirm that e2e treats all events with the same parameter signature as interchangeable.
I confirmed this by swapping places for `OwnershipTransferred` and `OwnershipTransferStarted` in various event assertions, and even creating an additional throwaway event to assert for instead, modifying their signatures on each test run and verifying that the tests pass only if the expected event parameter signature matches the emitted event's parameter signature, despite the name difference.
After digging a bit deeper into e2e, I soon realized the problem lay with `alloy-sol-types` - the version we're using (0.7.6) does not check event signatures when decoding logs, it only checks indexed parameters (topics[1..]) and data, and that the problem was fixed with [version 0.8.4](https://github.com/alloy-rs/core/releases/tag/v0.8.4).
Unfortunately, we're stuck with the older version stylus sdk uses...

In short, to solve the issue we will have to wait for Stylus SDK to start supporting alloy ^0.8.4 for this to be addressed.

### platform

- [X] linux
- [ ] windows
- [ ] macos

### Expected behavior

Event signatures should be taken into account when asserting that certain events were emitted.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

**Comment by 0xNeshi** [comment]
We're tracking new releases of https://github.com/OffchainLabs/stylus-sdk-rs that we depend on to address this

**Comment by bidzyyys** [comment]
@0xNeshi is it still blocked? With SDK `v0.7` we use new alloy.

**Comment by 0xNeshi** [comment]
@bidzyyys not blocked since #424 was merged!

**Comment by 0xNeshi** [comment]
Tried to repro the original issue, but it seems to be fixed now that Stylus is using alloy 0.8.14. Closing the issue.
